### ISSUE TITLE
res.locals is always empty

### ISSUE THREAD

**Issue by MvRemmerden** [FirebaseExtended/firebase-framework-tools#67]
The docs say: https://github.com/FirebaseExtended/firebase-framework-tools#optional-integrate-with-firebase-authentication

> The web framework-aware Firebase deployment tooling automatically keeps client and server state in sync using cookies. The Express res.locals object will optionally contain an authenticated Firebase App instance (firebaseApp) and the currently signed in user (currentUser).

However, when I try to get the res.locals in any deployment (either through Firebase Hosting, `firebase emulators:start` or `npm run dev:ssr`), the `res.locals` in `server.ts` is always empty. Is there anything I'm missing?

Also, it would be amazing if there could be a super minimal sample app that also shows how to then inject it into the module via the REQUEST token.

**Comment by MvRemmerden** [comment]
I have put together a minimal repository with authentication flow that based on everything I have read in the documentation and in the sample documents, should be working. I have injected the request and response in about every place I can imagine just to be safe, but none of them have a `locals` object.

https://github.com/MvRemmerden/firebase-universal-test

@jamesdaniels Is there a chance you might have a look and see what exactly I'm doing wrong here? I'm banging my head against the wall because I don't know how to debug this anymore than I have so far.

You can also see the latest deploy at https://fir-universal-test.web.app.

**Comment by gustavopch** [comment]
TL;DR: `npm i firebase`

---

I had the same issue and went ahead to investigate the source code.

`req.locals` is defined by `handleAuth` which is called by `handleFactory` at `src/firebase-aware.ts`:
https://github.com/FirebaseExtended/firebase-framework-tools/blob/8f0837005eb08e4ffd7c0ac41e8477420f8a8caf/src/firebase-aware.ts#L78-L79

`handleFactory` is only used when `usingFirebaseJsSdk` is truthy:
https://github.com/FirebaseExtended/firebase-framework-tools/blob/8f0837005eb08e4ffd7c0ac41e8477420f8a8caf/src/index.ts#L26-L31

`usingFirebaseJsSdk`'s value depends on `process.env.__FIREBASE_DEFAULTS__`:
https://github.com/FirebaseExtended/firebase-framework-tools/blob/8f0837005eb08e4ffd7c0ac41e8477420f8a8caf/src/index.ts#L19

And `process.env.__FIREBASE_DEFAULTS__` is defined by `firebase-tools` at https://github.com/firebase/firebase-tools/blob/master/src/frameworks/index.ts, but only when `@firebase/app`, which is a dependency of `firebase`, is installed.
